### PR TITLE
make StaticFileResponseContext constructable

### PR DESCRIPTION
**Pull request by SimonCropp**
so people can instantiate it for testing purposes

### CODE REVIEW

**Comment by SimonCropp** [comment]
arguably this is a breaking change. and i can re-add the extra empty constructor. but note that no one could do anything useful with the default contributor since all the props were internal. 

perhaps re-add the extra empty constructor with an obsolete?

File: src/Microsoft.AspNetCore.StaticFiles/StaticFileResponseContext.cs [hunk 1]
```diff
@@ -11,14 +12,28 @@ namespace Microsoft.AspNetCore.StaticFiles
     /// </summary>
     public class StaticFileResponseContext
     {
+        public StaticFileResponseContext(HttpContext context, IFileInfo file)
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.StaticFiles/StaticFileResponseContext.cs]: This is closer to the pattern we use in other repos like Security. Adding back the default constructor will avoid a binary break and avoid getting flagged by our breaking change tools (even if we didn't expect anyone to call that API).

Comment (SimonCropp) [reply src/Microsoft.AspNetCore.StaticFiles/StaticFileResponseContext.cs]: default added back in

**Comment by SimonCropp** [comment]
nah merge whenever. i was able to work around it by changing my method under test to take in HttpContext and IFileInfo instead. so whenever it is released i can revert the minor work around.

**Pull request merged**